src/Bounds.cpp

```diff
@@ -197,11 +194,11 @@ class Bounds : public IRVisitor {
             // Start with the bounds of the narrow type.
             bounds_of_type(from);
             // If we have a better min or max for the arg use that.
-            if (min_a.defined()) min = min_a;
-            if (max_a.defined()) max = max_a;
+            if (a.has_lower_bound()) interval.min = a.min;
```

Comment:
I feel there may be room for an Interval method to copy the bounds which are defined in one interval to another interval.